### ISSUE TITLE
Data loader

### ISSUE THREAD

**Issue by bdeterling** [bdeterling/axiom#3]

**Comment by jasonperrone** [comment]
Seems like a pretty big chunk of non-critical path work for the pilot/free phase.

**Comment by bdeterling** [comment]
Agree, let’s hold off. Although, we might need the backend part sooner rather than later if we are going to have any reports that use the generic data.

**Comment by jasonperrone** [comment]
I don't mind working on it ATP, but let me know if you have any higher priority stuff. I feel where I could lend the most value right now is in writing the reports that were needed for the pilot. Remember, this phase of the project should be doing the bare minimum it takes to get them to open their checkbooks. Otherwise, every second we spend is a waste of our life.

**Comment by bdeterling** [comment]
Yeah, unlike playing WoW which is so not a waste of your life.

**Comment by jasonperrone** [comment]
Well, it's all a matter of perspective. Lucky for you, Rails is damn near as fun as WoW.

**Comment by jasonperrone** [comment]
I think I want to support entities. In other words, when  allowing the user to define a new property, they can select the data type being 'string','number' or **entity**. If you choose entity, I can then present a list of the tables in the system and you have to tell me which one is your entity (so I can populate the external source field).  

Also, I think we should add a column to the point table when I can store some id to associate this data load.  In other words, when I load the dude's data into the database, I come up with some guid or something and slap in on each record and then tell the user what this id is. Why? So you can easily undo the load. Call it a data set ID. Let's say I load the data and I'm like FUCK, why did I do that? i can then input the dataset id and say "delete" and then I'll delete all the points I created from that import.

**Comment by bdeterling** [comment]
Yes, data type can be entity.

Agree about having a data load id.

Yes, the entity id will be on the entity, not on the type. I think the source should be on entity too although maybe there is a default on the entity type. That would give us support for hitting multiple systems like Staples.

**Comment by jasonperrone** [comment]
hmmm. yeah maybe.

**Comment by jasonperrone** [comment]
I think this is good for now. Put any enhancement requests in a separate issue.
